### PR TITLE
Issue 38 upload function

### LINKED ISSUE #69: #38upload function
1. Existing records in the database should NOT be replaced by new assignments. (Changes to the existing records should instead be made in the Edit page.)
2. If any duplicate workload(s) is being assigned, a message should be popped up to remind users of the fact that a workload(s) with the same details has been previously assigned.
3. For the purpose of MVP, we may assume that all users being assigned exist in the user table. Otherwise, a message should be popped up to remind users about the absence of assignee.

### PR DESCRIPTION
**Pull request by kylelhk**
## Change Summary 📝

1. Basic upload function without header and data type validation.
2. Existing records in the database are not replaced by new assignments. Changes to the existing records should instead be made in the Edit page.
3. For the purpose of MVP, we may assume that all users (Staff Number) being assigned exist in the user table. Otherwise, a message would pop up to remind users about the absence of assignee.
4. This function assumes that users have already confirmed the correctness of spreadsheet contents, so there is no check for duplicate workload(s).

---

## Change Form 📋
Please mark as NA if not applicable. If a certain criterion is not met, kindly provide a reason.

- [x] The pull request title has an issue number.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [ ] Relevant tests (if applicable) have been added or updated.

---

## Any background context you want to provide? 🖼️

1. The file to be uploaded must comprise a tab named "Work Assignment".
2. Two Excels were created for testing:
[testing.xlsx](https://github.com/yylocky/CITS5206-Group3/files/12916442/testing.xlsx)
[testing2.xlsx](https://github.com/yylocky/CITS5206-Group3/files/12924079/testing2.xlsx)

Open Issues / Bugs:
1. If the Staff Number of an assignment does not exist in the user table, any workload entry(ies) after that entry will not be processed.
2. Enhancement in file validation to be documented as future works (e.g. checking header and data type).

## Reviewers 🕵️

(Optional) Suggest reviewers that might be interested in this PR.

### CODE REVIEW

File: app/templates/assign_workload.html [hunk 1]
```diff
@@ -4,13 +4,13 @@
   rel="stylesheet"
   href="{{ url_for('static', filename='css/base.css') }}"
 />
-<link 
-rel="stylesheet" 
-href="{{ url_for('static', filename='css/view_workload.css') }}"
+<link
+  rel="stylesheet"
+  href="{{ url_for('static', filename='css/view_workload.css') }}"
 />
```
Comment (yylocky) [inline app/templates/assign_workload.html]: The stylesheet probably is not doing much work here.

File: app/routes.py [hunk 2]
```diff
@@ -98,6 +99,159 @@ def dashboard():
...
+                if item['Staff Number'] is None:
+                    continue
+
+                if item['Department'] is None:
+                    continue
+
+                user = User.query.filter_by(
+                    username=item['Staff Number']).first()
+                if user is None:
+                    flash("User {} does not exist.".format(
```
Comment (yylocky) [inline app/routes.py]: Can we use a more sensible message to let user know the previous rows have been processed?

**Review by yylocky — APPROVED** [review]
image
On top of the "file validation" addressed above,  insert value (ex. task type, calculation logic between leave, contract and available hours...) should also be addressed.

File: app/templates/assign_workload.html (hunk 1 above)
Comment (kylelhk) [reply app/templates/assign_workload.html]: Deleted

File: app/routes.py (hunk 2 above)
Comment (kylelhk) [reply app/routes.py]: Revised

**Comment by kylelhk** [comment]
> image
> On top of the "file validation" addressed above, insert value (ex. task type, calculation logic between leave, contract and available hours...) should also be addressed.

Documented in the Developer Guide.  Thanks.

**Pull request merged**